### PR TITLE
[resource based billing backend] Patches 1-5 add resource-based billing tables, migrations, methods

### PR DESCRIPTION
**Pull request by agreea**
## What Does this PR Do?

Implements the complete backend infrastructure for resource-based billing (patches 1-5), including database schemas for resources and resource claims, ORM methods with full test coverage, and Zod schema validation. Adds support for resource features and updates subscription item features to handle the new resource billing model.

**Key Changes:**
- New `resources` and `resource_claims` tables with proper indexing and constraints
- Resource and resource claim ORM methods with comprehensive test suite
- Resource feature types and subscription item feature support
- Database migration with FeatureType enum extension
- Fixed timestamp column handling and schema validation

**Tests:** All 26 resource-related tests passing, including edge cases for uniqueness constraints, external ID reuse, and pagination.

---
## Summary by cubic
Adds backend support for resource-based billing, including new resources and resource_claims models, a Resource feature type, and full ORM/Zod coverage. Enables attaching resources to subscription item features and claiming/releasing them with safe constraints.

- New Features
  - Added resources and resource_claims tables with FKs, RLS, and targeted indexes (including partial unique on active external_id).
  - Extended FeatureType with resource and added resourceId to features and subscription item features.
  - Implemented ORM methods and Zod schemas for resources and claims (select/insert/update, bulk, cursor pagination, upsert by pricingModelId+slug).
  - Added claim helpers: select active claims, count active claims per feature, release single/all claims, find active by externalId.
  - Seed helpers for resource features, resources, and claims; UI shows “Resource” in SubscriptionFeaturesTable.
  - Comprehensive tests for resources, resource claims, features, and subscription item features.
  - Performance and safety improvements: indexes on resourceId FKs, DB-level filtering and COUNT for active claims, create-only FK columns to prevent updates, and stricter customer read policy scoped to pricing models.

- Migration
  - Run drizzle migration 0266 to create new tables, policies, indexes, and extend the FeatureType enum.
  - No data backfill required; existing features remain unchanged.

<sup>Written for commit def1d2c1af1f3f0146d4377b0096c6d48fb32ff3. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Added Resource as a new feature type with full resource management (create, track, claim, release) and subscription support.
  * Public APIs and schemas updated to handle resource-based features and subscription item resource links.
  * UI: feature type label and table rendering updated to surface Resource entries.

* **Tests**
  * Comprehensive tests added for resource CRUD, claims lifecycle, subscription-item resource behavior, and related validations.

* **Migrations**
  * Database migration added to introduce resources and resource-claims tables and indexes.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/db/tableMethods/resourceMethods.test.ts [hunk 1]
```diff
@@ -0,0 +1,714 @@
...
+        expect(page1.hasNextPage).toBe(true)
+
+        // All items should have pricing model info
+        for (const item of page1.items) {
+          expect(item.pricingModel.id).toBeTruthy()
+          expect(item.pricingModel.name).toBeTruthy()
+        }
+
+        // Get second page using cursor
+        if (page1.endCursor) {
```
[the remark this answers is not in the document]
Comment (BrooksFlannery) [reply platform/flowglad-next/src/db/tableMethods/resourceMethods.test.ts]: +1

**Review by BrooksFlannery — APPROVED** [review]
lgtm
